src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -333,22 +333,23 @@ protected override Expression VisitBlock(BlockExpression node)
 
         protected override Expression VisitGoto(GotoExpression node)
         {
-            Type type = node.Value.Type;
+            Type type = node.Value!.Type;
             if (!typeof(IQueryable).IsAssignableFrom(type))
                 return base.VisitGoto(node);
             LabelTarget target = VisitLabelTarget(node.Target);
             Expression value = Visit(node.Value);
             return Expression.MakeGoto(node.Kind, target, value, GetEquivalentType(typeof(EnumerableQuery).IsAssignableFrom(type) ? value.Type : type));
         }
 
-        protected override LabelTarget VisitLabelTarget(LabelTarget node)
+        [return: NotNullIfNotNull("node")]
+        protected override LabelTarget? VisitLabelTarget(LabelTarget? node)
```

Comment:
NIT: seems return type doesn't have to be nullable, then we wouldn't need `[return: NotNullIfNotNull("node")]`